File: prime_number.js


const isPrimeNumber = (number) => {

    for (let i = 2; i < Math.sqrt(number); i++) {

        if ((number % i) === 0) {
            return false;
        }


    }
    return true;
}

console.log(isPrimeNumber(53))


const LCM = (number1, number2) => {

    if (number2 <= 0) {
        return number1
    }


    return LCM(number2, number1 % number2);
}

console.log(LCM(100, 12))



const reverseNumber = (number) => {
    let lastDigit = 0;
    let rev = 0;

    while (number != 0) {

        lastDigit = num % 10;
        rev = rev * 10 + lastDigit;
        number = Math.floor(number)
    }

    return number;
}

File: bin/prime_number.dart
import 'dart:math' as math;

bool isPrimeNumber(int number) {
  for (int i = 2; i < math.sqrt(number); i++) {
    if ((number % i) == 0) {
      return false;
    }
  }

  return true;
}

// euler formula
int HCF(int number1, int number2) {
  if (number2 == 0) {
    return number1;
  }
  return HCF(number2, number1 % number2);
}

int reverseNumber(number) {
  int reverse = 0;

  while (number > 0) {
    int lastNumber = number % 10;
    reverse = reverse * 10 + lastNumber;
    number = number ~/ 10;
  }
  return reverse;
}

void main() {
  print(reverseNumber(12345));
}
